samples/detect_intent_with_sentiment_analysis.py

```diff
@@ -0,0 +1,105 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dialogflow API Beta Detect Intent Python sample with sentiment analysis.
+
+Examples:
+  python detect_intent_with_sentiment_analysis.py -h
+  python detect_intent_with_sentiment_analysis.py --project-id PROJECT_ID \
+  --session-id SESSION_ID \
+  "hello" "book a meeting room" "Mountain View"
+"""
+
+import argparse
+import uuid
+
+
+# [START dialogflow_detect_intent_with_sentiment_analysis]
+def detect_intent_with_sentiment_analysis(project_id, session_id, texts,
+                                          language_code):
+    """Returns the result of detect intent with texts as inputs and analyzes the
+    sentiment of the query text.
+
+    Using the same `session_id` between requests allows continuation
+    of the conversaion."""
+    import dialogflow_v2beta1 as dialogflow
+    session_client = dialogflow.SessionsClient()
+
+    session = session_client.session_path(project_id, session_id)
+    print('Session path: {}\n'.format(session))
+
+    for text in texts:
+        text_input = dialogflow.types.TextInput(
+            text=text, language_code=language_code)
+
+        query_input = dialogflow.types.QueryInput(text=text_input)
+
+        # Enable sentiment analysis
+        sentiment_config = dialogflow.types.SentimentAnalysisRequestConfig(
+            analyze_query_text_sentiment=True)
+
+        # Set the query parameters with sentiment analysis
+        query_params = dialogflow.types.QueryParameters(
+            sentiment_analysis_request_config=sentiment_config)
+
+        response = session_client.detect_intent(
+            session=session, query_input=query_input,
+            query_params=query_params)
+
+        print('=' * 20)
+        print('Query text: {}'.format(response.query_result.query_text))
+        print('Detected intent: {} (confidence: {})\n'.format(
+            response.query_result.intent.display_name,
+            response.query_result.intent_detection_confidence))
+        print('Fulfillment text: {}\n'.format(
+            response.query_result.fulfillment_text))
+        # Score between -1.0 (negative sentiment) and 1.0 (positive sentiment).
+        print('Query Text Sentiment Score: {}\n'.format(
+            response.query_result.sentiment_analysis_result
+            .query_text_sentiment.score))
+        print('Query Text Sentiment Magnitude: {}\n'.format(
+            response.query_result.sentiment_analysis_result
+            .query_text_sentiment.magnitude))
```

Comment:
suppose a language API call error happened. in this case, we'd get diagnostic_info back with key "query_sentiment_error", and the sentiment_analysis_result would not be present in the response.

i am not familiar with python protos, but will they just use default val? In other words, in presence of error, would it just print 0 for score and magnitude? If so, that would not be technically correct. It should only print 0, meaning neutral, when a successful response was returned. if an error happened it should probably say "error blah"

Reply:
indeed the sample as is would probably just print the default values in case of an error, but the error probably should be raised by the client library.

in any case, we would prefer to keep the samples more focused on how to call the service, as opposed to making it fully robust at the cost of readability for developers who are learning how to use the service or a new feature.